packages/core/src/lib/lit-core.ts

```diff
@@ -220,41 +220,85 @@ export class LitCore {
     return globalThis.logManager.getLogsForId(id);
   };
 
-  private async _getValidatorData() {
-    const [minNodeCount, bootstrapUrls] = await Promise.all([
-      LitContracts.getMinNodeCount(
-        this.config.litNetwork,
-        this.config.contractContext,
-        this.config.rpcUrl
-      ),
-      LitContracts.getValidators(
-        this.config.litNetwork,
-        this.config.contractContext,
-        this.config.rpcUrl,
-        this.config.nodeProtocol
-      ),
-    ]);
+  /**
+   * Retrieves the validator data including staking contract, epoch, minNodeCount, and bootstrapUrls.
+   * @returns An object containing the validator data.
+   * @throws Error if minNodeCount is not provided, is less than or equal to 0, or if bootstrapUrls are not available.
+   */
+  private async _getValidatorData(): Promise<{
+    stakingContract: ethers.Contract;
+    epoch: number;
+    minNodeCount: number;
+    bootstrapUrls: string[];
+  }> {
+    const { stakingContract, epoch, minNodeCount, bootstrapUrls } =
+      await LitContracts.getConnectionInfo({
+        litNetwork: this.config.litNetwork,
+        networkContext: this.config.contractContext,
+        rpcUrl: this.config.rpcUrl,
+        nodeProtocol: this.config.nodeProtocol,
+      });
+
+    // Validate minNodeCount
+    if (minNodeCount === undefined || minNodeCount === null) {
+      throw new Error('minNodeCount is required');
+    }
 
-    if (minNodeCount <= 0) {
+    const minNodeCountInInt = parseInt(minNodeCount.toString());
+
+    if (isNaN(minNodeCountInInt) || minNodeCountInInt <= 0) {
       throwError({
         message: `minNodeCount is ${minNodeCount}, which is invalid. Please check your network connection and try again.`,
         errorKind: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.kind,
         errorCode: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.name,
       });
     }
 
-    if (bootstrapUrls.length <= 0) {
+    // Validate bootstrapUrls
+    if (!Array.isArray(bootstrapUrls) || bootstrapUrls.length <= 0) {
       throwError({
         message: `Failed to get bootstrapUrls for network ${this.config.litNetwork}`,
         errorKind: LIT_ERROR.INIT_ERROR.kind,
         errorCode: LIT_ERROR.INIT_ERROR.name,
       });
     }
 
+    // Validate stakingContract
+    if (!stakingContract) {
+      throwError({
+        message: 'stakingContract is required',
+        errorKind: LIT_ERROR.INIT_ERROR.kind,
+        errorCode: LIT_ERROR.INIT_ERROR.name,
+      });
+    }
+
+    // Validate epoch
+    if (epoch === undefined || epoch === null) {
+      throwError({
+        message: 'epoch is required',
+        errorKind: LIT_ERROR.INIT_ERROR.kind,
+        errorCode: LIT_ERROR.INIT_ERROR.name,
+      });
+    }
+
+    const epochInInt = parseInt(epoch.toString());
```

Comment:
we might want to specify a radix in this case since we might eventually change the numeric base. So perhaps we explicitly specify `base 10`